Mathlib/Algebra/NonAssoc/PreLie/Basic.lean

```diff
@@ -0,0 +1,122 @@
+/-
+Copyright (c) 2025 Nikolas Tapia. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nikolas Tapia
+-/
+import Mathlib.Algebra.Module.Opposite
+import Mathlib.Algebra.Ring.Associator
+import Mathlib.GroupTheory.GroupAction.Ring
+/-!
+# Pre-Lie rings and algebras
+
+In this file we introduce left and right pre-Lie rings, defined as a `NonUnitalNonAssocRing` where
+the associator `associator x y z := (x * y) * z - x * (y * z)` is left or right symmetric,
+respectively.
+
+We prove that every `Left(Right)PreLieRing L` is a `Right(Left)PreLieRing L` with
+the opposite `mul`. The equivalence is simple given by `op : L ≃* Lᵐᵒᵖ`.
+
+Everything holds for the algebra versions where `L` is also an `R-Module` over a commutative ring
+`R`.
+
+## Main definitions
+  All are a defined as a `NonUnitalNonAssocRing` whose `associator` satisfies an identity.
+  * `LeftPreLieRing`
+  * `RightPreLieRing`
+  * `LeftPreLieAlgebra`
+  * `RightPreLieAlgebra`
+
+## Main results
+  * Every `LeftPreLieRing` is a `RightPreLieRing` with the opposite multiplication.
+
+## Implementation notes
+There are left and right versions of the structures, equivalent via `ᵐᵒᵖ`.
+Perhaps one could be favored but there is no real reason to.
+
+## Tags
+pre-lie algebras
+
```

Comment:
Some old files still contain these `Tags` but we gave up on them (and are likely to cull them all some day) so I suggest dropping them here.
```suggestion
```